### PR TITLE
Bugfix - "Apply" button doesn't work with dynamic content in emails

### PR DESCRIPTION
**Pull request by virlatinus**
| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/4963, https://github.com/mautic/mautic/issues/5120, https://github.com/mautic/mautic/issues/4971, https://github.com/mautic/mautic/issues/3921
| BC breaks? | N/A
| Deprecations? | N/A

#### Description:
This PR fixes an issue with Dynamic Email Content (DEC) slots converting to tokens when clicking the apply button.
It also fixes issues with Dynamic Content in Code Mode: the add variants and delete DEC buttons are back. Dynamic Content items can be deleted (except for the first one because there should be at least one).

#### Steps to reproduce the bug:
1. Create/Edit an email
2. Open the builder
2. Add DEC slots and enter some content
3. Click the Apply button
4. The DEC slots will disappear and tokens will be in their place.
5. Close the builder
6. Switch to code mode theme
7. You can add new DECs, but you can't delete them. 
8. You can't add variants either.

#### Steps to test this PR:
1. Apply this PR
2. Try to replicate the issues
3. The DEC slots will remain in place.
4. Switch to Code Mode, you can add/delete DECs and variants

### CODE REVIEW

File: app/bundles/CoreBundle/Form/Type/DynamicContentFilterType.php [hunk 1]
```diff
@@ -34,6 +34,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'attr'  => [
                     'class' => 'form-control dynamic-content-token-name',
                 ],
+                'read_only' => true,
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Form/Type/DynamicContentFilterType.php]: Did you mean to make this read only? JS won't be able to update this dynamically if set as so.

**Comment by escopecz** [comment]
It all works for me, but I don't get one thing. I have 2 DEC slots in the builder, but if I switch to the code mode, there are 3. What's the reason? How do I edit the first one in the builder?
dec-in-builder
dec-in-codemode

**Comment by virlatinus** [comment]
@escopecz the first one is the prototype from which the other DECs are created. It can be modified and used in Code Mode, but not in the builder.

**Comment by escopecz** [comment]
Shouldn't be the prototype hidden from UI? Or does it have some functional advantage in the code mode?

I think this can be merged. My questions are just out of curiosity. I know it works the same way on staging too, so no change in that area.

**Pull request merged**